src/ImageSharp/Formats/DecoderOptions.cs

```diff
@@ -15,15 +15,21 @@ public sealed class DecoderOptions
 
     private uint maxFrames = int.MaxValue;
 
+    // Used by the FileProvider in the unit tests to set the configuration on the fly.
+#pragma warning disable SA1401 // Fields should be private
+    internal Configuration BackingConfiguration = Configuration.Default;
+#pragma warning restore SA1401 // Fields should be private
```

Comment:
Why not a member `SetConfiguration()` method and a private field?

Reply:
Yeah... I was trying to avoid adding to the method itself and overcomplicated things. I moved it but still had to suppress a warning as the IDE thinks incorrectly that it can be an auto-implemented property.